src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs

```diff
@@ -371,80 +376,182 @@ public virtual string GetFixAllTitle(FixAllState fixAllState)
+                //  1) if both changes are the same.
+                //  2) the change we're hitting is not empty and we're at the end of the other change.
+
+                // Specifically, it is not safe for us to insert somewhere in start-to-middle of an 
+                // existing non-empty change.  And if we have another *empty* change, then it's 
+                // not safe for both of us to be inserting at the same point (except when the 
+                // change is identical).
+                var otherChange = intersectingSpans[0];
+                if (otherChange == change)
+                {
+                    // We're both inserting the same text at the same position.  This is safe.
+                    return true;
+                }
+
+                return !otherChange.Span.IsEmpty &&
+                       otherChange.Span.End == change.Span.Start;
+            }
+
+            // We're intersecting multiple changes.  That's never OK.
+            return false;
+        }
+
+        private static bool OverwriteChangeCanBeApplied(
+            TextChange change,
+            ArrayBuilder<TextChange> overlappingSpans,
+            ArrayBuilder<TextChange> intersectingSpans)
+        {
+            Debug.Assert(!change.Span.IsEmpty);
+
+            return !OverwriteChangeConflictsWithOverlappingSpans(change, overlappingSpans) &&
+                   !OverwriteChangeConflictsWithIntersectingSpans(change, intersectingSpans);
+        }
+
+        private static bool OverwriteChangeConflictsWithOverlappingSpans(
+            TextChange change,
+            ArrayBuilder<TextChange> overlappingSpans)
+        {
+            Debug.Assert(!change.Span.IsEmpty);
+
+            if (overlappingSpans.Count == 0)
+            {
+                // This change didn't overlap with any other changes.  This change is safe to make.
+                return false;
+            }
+
+            // The change we want to make overlapped an existing change we're making.  Only allow
+            // this if there was a single overlap and we are exactly the same change as it.
+            // Otherwise, this is a conflict.
+            return overlappingSpans.Count > 1 || overlappingSpans[0] != change;
+        }
+
+        private static bool OverwriteChangeConflictsWithIntersectingSpans(
+            TextChange change,
+            ArrayBuilder<TextChange> intersectingSpans)
+        {
+            Debug.Assert(!change.Span.IsEmpty);
+
+            // We care about our intersections with *empty-span* changes.  Non-empty changes that
+            // we overlap are already handled in OverwriteChangeConflictsWithOverlappingSpans.
+            // And non-empty spans that we abut (i.e. which we're adjacent to) are totally safe 
+            // for both to be applied.
+            //
+            // However, empty-span changes are extremely ambiguous. It is not possible to tell which
```

Comment:
"empty-span changes" is a weird term. Can you just better define "pure insertion" at the beginning and then rely on that term instead?